compiler-rt/lib/asan/asan_globals.cpp

```diff
@@ -225,10 +248,12 @@ static void RegisterGlobal(const Global *g) {
   }
   if (CanPoisonMemory())
     PoisonRedZones(*g);
-  ListOfGlobals *l = new (GetGlobalLowLevelAllocator()) ListOfGlobals;
-  l->g = g;
-  l->next = list_of_all_globals;
-  list_of_all_globals = l;
+
+  AddGlobalToList(list_of_all_globals, g);
+
+  if (UseODRIndicator(g) && g->odr_indicator != UINTPTR_MAX)
```

Comment:
I'd prefer we do this in CheckODRViolationViaIndicator, 
so instead of 2 lookups: find + [], 
just one one [] there.

Reply:
I can do that. The only concern is that `Check` will be more like `CheckAndAdd` rather than just doing checking. OTOH we won't need to repeat this guard.

Reply:
I don't have problem with the current name, we do the `check`, and adding into the map is just `check` implementation detail.

Reply:
@vitalybuka I moved things around. LMK if now looks good.